README.md

```diff
@@ -415,16 +415,93 @@ EXPECT_EQ("(A+(B+C))",
           (bricks::metaprogramming::reduce<concatenate_s, std::tuple<A, B, C>>::s()));
   
 // Combine.
-struct A { static std::string foo() { return "foo"; } };
-struct B { static std::string bar() { return "bar"; } };
-struct C { static std::string baz() { return "baz"; } };
-
-bricks::metaprogramming::combine<std::tuple<A, B, C>> c;
+struct NEG {
+  // A simple way to differentiate logic by struct/class type
+  // is to provide a local, unique, symbol as the 1st param.
+  struct TYPE {};
+  // Combine-able operations are defined as `operator()`.
+  // Just because we need to pick one common name,
+  // otherwise more `using`-s will be needed.
+  int operator()(TYPE, int a) { return -a; }
+};
 
-EXPECT_EQ("foo", c.foo());
-EXPECT_EQ("bar", c.bar());
-EXPECT_EQ("baz", c.baz());
+struct ADD {
+  struct TYPE {};
+  int operator()(TYPE, int a, int b) { return a + b; }
+  // Prove that the method is instantiated at compile time.
+  template <typename T>
+  int operator()(TYPE, int a, int b, T c) {
+    return a + b + AsInt(c);
+  }
+};
 
+// Since "has-a" is used instead of "is-a",
+// mutual inheritance of underlying types
+// is not a problem at all.
+// Confirm this by making `MUL` inherit from `ADD`.
+struct MUL : ADD {
+  struct TYPE {};
+  int operator()(TYPE, int a, int b) { return a * b; }
+  int operator()(TYPE, int a, int b, int c) { return a * b * c; }
+};
+  
+// User-friendly method names, internally dispatching calls via `operator()`.
+// A good way to make sure new names appear in one place only, since
+// using `using`-s would require writing them down at least twice each.
```

Comment:
Somewhat -- I make comments shorter for a Quora post / Gist earlier, and then realized our Readme could also benefit from short comments.

Let's tackle this after we merge these few pending changes -- they are becoming harder and harder to manage.